src/app/convai-checker.component.spec.ts

```diff
@@ -278,6 +279,172 @@ function verifyEmojiIconsInDomWithZeroOpacity() {
   expect(getElementOpactiy('sadEmoji')).toEqual(0);
 }
 
+// TODO(rachelrosen): Refactor this into smaller functions.
+function verifyUILayerChangesForDemoConfig(
```

Comment:
This function name isn't that helpful, all I get from it is there are going to be some changes to the UI and that they should be verified.  Breaking this up into specifically "X will change and then I expect Y to happen" would be more useful in my opinion.  Don't treat this as blocking though, completely get that testing just is terrible.

Reply:
I tried to make the name better; this is checking how the "layers" of the UI (the bit that shows the score information, the screen that appears after the user clicks "Seem wrong?", and the "Thanks for the feedback" screen) transition from one state to the other from user interaction via clicking or typing. Layers are a conceptual thing related to the design of this widget, and I realize now that it wasn't clear without the right amount of context. I also added a comment.